<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Principal</class>
 <widget class="QMainWindow" name="Principal">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1040</width>
    <height>720</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>285</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1040</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Arvore Mínima de Steiner</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="gb_grafo">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>191</width>
      <height>261</height>
     </rect>
    </property>
    <property name="title">
     <string>Problema:</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QComboBox" name="cmb_grafo">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>171</width>
       <height>25</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_grafo">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Instância:</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="chk_previsualizar">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>171</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Pré-visualizar</string>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>110</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="frameShape">
      <enum>QFrame::NoFrame</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="text">
      <string>Problema:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>Solução:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_qtd_vert">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>130</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Qtd. Pts. Fixos:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_mst">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>190</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>MST:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_smt">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>210</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>SMT:</string>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_qtd_ptst">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Qtd. Pts. Steiner:</string>
     </property>
    </widget>
    <widget class="Line" name="line_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>171</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_reducao">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>230</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Redução:</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tb_solucao">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>10</y>
      <width>821</width>
      <height>651</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_grafo">
     <attribute name="title">
      <string>Grafo</string>
     </attribute>
     <widget class="QGraphicsView" name="grph_grafo">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>800</width>
        <height>600</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QAbstractScrollArea::AdjustIgnored</enum>
      </property>
      <property name="interactive">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_solucao">
     <attribute name="title">
      <string>GA</string>
     </attribute>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Grafo:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Qtd. Indivíduos:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Cruzamento:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Mutaçao:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>230</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Qtd. Gerações:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>420</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>Fitness:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_grafo">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_qtdInd">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_muta">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_cruza">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>150</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_qtdGera">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>250</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_fitness">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>440</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>10</y>
        <width>20</width>
        <height>601</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_semRnd">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>300</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="saida_semRnd">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>280</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>Semente Aleat.:</string>
      </property>
     </widget>
     <widget class="Line" name="line_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>111</width>
        <height>20</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>101</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>Parametros:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>350</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>75</weight>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>Solução:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_qtdPtSt">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>390</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>370</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>Qtd. Pts. Steiner:</string>
      </property>
     </widget>
     <widget class="QCustomPlot" name="plt_grafico" native="true">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>10</y>
        <width>671</width>
        <height>601</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_tempo">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>540</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>520</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>Tempo de Exec.:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="txt_saida_reducao">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>490</y>
        <width>111</width>
        <height>25</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>470</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="text">
       <string>% de Redução:</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="gb_ams">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>191</width>
      <height>381</height>
     </rect>
    </property>
    <property name="title">
     <string>Algoritmo Genetico:</string>
    </property>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Qtd. Indivíduos:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Cruzamento (%):</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Mutação (%):</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>180</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Qtd. Gerações:</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="spn_ams_individuos">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>71</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <number>2</number>
     </property>
     <property name="maximum">
      <number>9999</number>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spn_ams_cruzamento">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>81</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>100.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="spn_ams_mutacao">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>81</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>100.000000000000000</double>
     </property>
    </widget>
    <widget class="QSpinBox" name="spn_ams_geracoes">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>111</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>900000</number>
     </property>
    </widget>
    <widget class="QLabel" name="lbl_semente">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>230</y>
       <width>171</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Semente Aleatória:</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="txt_random">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>250</y>
       <width>171</width>
       <height>25</height>
      </rect>
     </property>
     <property name="inputMethodHints">
      <set>Qt::ImhDigitsOnly</set>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="maxLength">
      <number>11</number>
     </property>
    </widget>
    <widget class="QPushButton" name="btn_iniciar">
     <property name="enabled">
      <bool>false</bool>
     </property>
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>171</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Iniciar</string>
     </property>
    </widget>
   </widget>
   <zorder>gb_ams</zorder>
   <zorder>gb_grafo</zorder>
   <zorder>tb_solucao</zorder>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1040</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuArquivo">
    <property name="title">
     <string>Arquivo</string>
    </property>
    <addaction name="act_abrir_solucao"/>
    <addaction name="separator"/>
    <addaction name="act_fechar"/>
   </widget>
   <widget class="QMenu" name="menuAjuda">
    <property name="title">
     <string>Ajuda</string>
    </property>
    <addaction name="act_sobre"/>
   </widget>
   <addaction name="menuArquivo"/>
   <addaction name="menuAjuda"/>
  </widget>
  <action name="act_abrir_solucao">
   <property name="text">
    <string>Abrir Solução</string>
   </property>
  </action>
  <action name="act_fechar">
   <property name="text">
    <string>Fechar</string>
   </property>
  </action>
  <action name="act_sobre">
   <property name="text">
    <string>Sobre</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
